skimage/morphology/convex_hull.py

```diff
@@ -126,7 +126,8 @@ def convex_hull_object(image, neighbors=8):
     if neighbors != 4 and neighbors != 8:
         raise ValueError('Neighbors must be either 4 or 8.')
 
-    labeled_im = label(image, neighbors, background=0)
+    connectivity = neighbors // 4
+    labeled_im = label(image, connectivity=connectivity, background=0)
```

Comment:
Happy to leave this to another PR, but we should deprecate neighbors here also. Let me know if you want to leave it for later.

Reply:
Let's leave it for later